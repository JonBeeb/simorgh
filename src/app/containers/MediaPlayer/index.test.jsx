import { shouldMatchSnapshot, isNull } from '@bbc/psammead-test-helpers';
import {
  VideoCanonical,
  VideoAmp,
  VideoCanonicalNoVersionId,
  VideoCanonicalToggledOff,
  VideoCanonicalWithCaption,
  VideoAmpWithCaption,
} from './fixtureData';

describe('MediaPlayer', () => {
  shouldMatchSnapshot(
    'Calls the canonical placeholder when platform is canonical',
    VideoCanonical,
  );

  shouldMatchSnapshot('Calls the AMP player when platform is AMP', VideoAmp);

  shouldMatchSnapshot(
    'Calls the canonical player with a caption',
    VideoCanonicalWithCaption,
  );

  shouldMatchSnapshot(
    'Calls the AMP player with a caption',
    VideoAmpWithCaption,
  );

  describe('Fails and returns early when', () => {
    isNull('there is no versionId', VideoCanonicalNoVersionId);
    isNull('component is toggled off', VideoCanonicalToggledOff);
  });
});
