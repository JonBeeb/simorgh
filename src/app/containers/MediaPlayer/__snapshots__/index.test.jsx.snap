// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer Calls the AMP player when platform is AMP 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  grid-column: 1 / span 6;
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c0 {
    max-width: initial;
  }
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <amp-iframe
            allowfullscreen="allowfullscreen"
            frameborder="0"
            layout="fill"
            sandbox="allow-scripts allow-same-origin"
            src="https://www.test.bbc.com/ws/av-embeds/articles/foo/p01k6msp/en-GB/amp"
          >
            <amp-img
              layout="fill"
              placeholder="true"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
            />
          </amp-iframe>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Calls the AMP player with a caption 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c3 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c2 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  color: #6E6E73;
  padding: 0.5rem 0.5rem 0;
  width: 100%;
}

.c2 > p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.c2 > p:last-child {
  padding-bottom: 0;
}

.c2 i {
  font-style: normal;
}

.c0 {
  grid-column: 1 / span 6;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c2 {
    padding: 0.5rem 1rem 0;
  }
}

@media (min-width:63rem) {
  .c2 {
    padding: 0.5rem 0 0;
  }
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c0 {
    max-width: initial;
  }
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <amp-iframe
            allowfullscreen="allowfullscreen"
            frameborder="0"
            layout="fill"
            sandbox="allow-scripts allow-same-origin"
            src="https://www.test.bbc.com/ws/av-embeds/articles/foo/p01k6msp/en-GB/amp"
          >
            <amp-img
              layout="fill"
              placeholder="true"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
            />
          </amp-iframe>
        </div>
        <figcaption
          class="c2"
        >
          <span
            class="c3"
          >
            Video caption, 
          </span>
          <p>
            Media Player With Caption
          </p>
        </figcaption>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Calls the canonical placeholder when platform is canonical 1`] = `
.c3 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c2 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  grid-column: 1 / span 6;
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c0 {
    max-width: initial;
  }
}

<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <img
              alt="Image Alt"
              class="c3"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
            />
          </div>
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Calls the canonical player with a caption 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    .c3 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c2 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c1 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c5 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  color: #6E6E73;
  padding: 0.5rem 0.5rem 0;
  width: 100%;
}

.c4 > p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.c4 > p:last-child {
  padding-bottom: 0;
}

.c4 i {
  font-style: normal;
}

.c0 {
  grid-column: 1 / span 6;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c4 {
    padding: 0.5rem 1rem 0;
  }
}

@media (min-width:63rem) {
  .c4 {
    padding: 0.5rem 0 0;
  }
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c0 {
    max-width: initial;
  }
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <img
              alt="Image Alt"
              class="c3"
              src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
            />
          </div>
        </div>
        <figcaption
          class="c4"
        >
          <span
            class="c5"
          >
            Video caption, 
          </span>
          <p>
            Media Player With Caption
          </p>
        </figcaption>
      </div>
    </div>
  </body>
</html>
`;
